### PR TITLE
feature(locksmith): Natively generate Google Wallet pass for lock's key

### LINKED ISSUE #13855: Native iOS and Android Passes API
Our application includes the ability for owners of any NFT to add it as a "pass" on their Google or iOS devices.

For now, these passes were generated using https://ethpass.xyz/. Last year EthPass was acquired by Moonpay and it's unclear how long support for these will remain so we want to bring that capability in-house.

As such we will need to introduce a new Locksmith API that generates passes for iOS or Android. It is ok to use 3rd party libraries but we should not depend on 3rd party APIs, outside of Apple's and Google's APIs.

The API will consist of 2 `GET` endpoints:
* `/v2/pass/{network}/{lockAddress}/{keyId}/ios` and,
* `/v2/pass/{network}/{lockAddress}/{keyId}/android` 

These endpoints will respectively create the iOS and Android passes for the key `keyId` of the lock `lockAddress` on the network `network`. Importantly the pass will include the QR code signed by Locksmith so that they can be used to "store" event tickets.

The design of the passes should include the following elements and is left up to the implementer:
- key/NFT image
- key id
- lock address
- lock name
- network name
- Unlock Protocol logo

Once the API is ready it will need to be added to Unlock-app and replace the ethpass API calls!

We are attaching a 2000 USDC bounty for this feature which needs to be fully tested!

### PR DESCRIPTION
**Pull request by 0xTxbi**
# Description

This PR replaces third-party dependencies with an in-house solution for generating Google Wallet passes for a lock's key owners. 

---
#### Changes Implemented

**Configuration:**

- **Details:** Centralized configuration for Google API credentials and settings.
- **Implementation:**
  - Loaded, and parsed credentials from environment variables.

**Authentication:**

**Details:** Initialized a `GoogleAuth` client for authenticated API requests.
- **Implementation:** 
  - Configured `googleAuthClient` using `GoogleAuth` with credentials and scopes from `config`.

**Pass Class Management:**

- **Details:** Functions to check and create Google Wallet pass classes.
- **Implementation:**
  - `getOrCreateWalletClass`: Verifies existence of a pass class, and creates one if it doesn't exist.

**Pass Object Creation:**

- **Details:** Function to create Google Wallet pass objects.
- **Implementation:**
  - Pass objects include lock metadata and verification QR code.

**API Endpoint:**

- **Details:** New endpoint for generating Android passes.
- **Implementation:**
  - Endpoint: `/v2/pass/{network}/{lockAddress}/{keyId}/android`.
  - Handles parameter extraction, metadata retrieval, and pass generation.

**Error Handling and Logging:**

- **Details:** Robust error handling and logging.
- **Implementation:**
  - Try-catch blocks with detailed logging for debugging and maintenance.

---

#### Integration Workflow

1. **Configuration and Authentication:**
   - Load settings and initialize `googleAuthClient`.

2. **Pass Class Management:**
   - Check for pass class existence or create if necessary.

3. **Pass Object Creation:**
   - Retrieve lock metadata.
   - Generate verification QR code.
   - Create pass object.

4. **API Endpoint:**
   - Extract request parameters.
   - Ensure pass class existence.
   - Generate and return pass object URL.

---

#### Setup Instructions

**Environment Variables:**

- `GOOGLE_APPLICATION_CREDENTIALS`: Path to Google API credentials JSON file.
  - **Obtain from:** Google Cloud Console, Service Accounts.
- `GOOGLE_WALLET_API_ISSUER_ID`: Issuer ID for Google Wallet API.
  - **Obtain from:** Google Wallet API issuer account setup.
- `GOOGLE_WALLET_API_CLASS`: Desired identifier for wallet class.

**Example .env File:**

```env
GOOGLE_APPLICATION_CREDENTIALS=/path/to/credentials.json
GOOGLE_WALLET_API_ISSUER_ID=your_issuer_id
GOOGLE_WALLET_API_CLASS=your_desired_wallet_class
```

**Steps to Obtain Values:**

1. **Google API Credentials:**
   - Navigate to Google Cloud Console.
   - Create/download JSON key from Service Accounts.

2. **Google Wallet API Issuer ID:**
   - Set up an issuer account in Google Wallet API.
   - Retrieve issuer ID.

---

#### Screengrabs

1. **Add Wallet Pass Page:**
add-google-wallet-pass-intro

   - Description: Showing the prompt to add a ticket for the lock.

2. **Success State After Adding Wallet Pass:**
add-google-wallet-pass-success

   - Description: Displaying the success state after adding the wallet pass.

3. **Wallet Pass in Google Wallet App:**
saved-google-wallet-pass-view

   - Description: The wallet pass as displayed in the Google Wallet app.

# Issues

Fixes #13855
Refs #13855 

# Checklist:

- [x] 1 PR, 1 purpose: my Pull Request applies to a single purpose
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the docs to reflect my changes if applicable
- [ ] I have added tests (and stories for frontend components) that prove my fix is effective or that my feature works
- [x] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

## Release Note Draft Snippet

### CODE REVIEW

File: locksmith/package.json [hunk 1]
```diff
@@ -98,6 +100,7 @@
     "@types/cors": "2.8.17",
     "@types/geoip-country": "4.0.2",
     "@types/isomorphic-fetch": "0.0.39",
+    "@types/jsonwebtoken": "^9",
```
Comment (julien51) [inline locksmith/package.json:103]: Please use a specific version here to ensure consistent builds

File: locksmith/src/operations/generate-pass/android/createPassObject.ts [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+  lockName: string,
+  networkName: string,
+  lockAddress: string,
+  keyId: string,
+  qrCodeUrl: string
+) {
+  //   create a new wallet pass for the user
+  const objectId = `${classId}.${keyId}`
+
+  console.log(objectId)
```
Comment (julien51) [inline locksmith/src/operations/generate-pass/android/createPassObject.ts:19]: cleanup :)

**Review by julien51 — APPROVED** [review]
ok that looks good. I would love to see some unit tests here as well, especially to ensure that we have the right "format".
I will set the env vars but please make sure you fix CI here

File: locksmith/package.json (hunk 1 above)
Comment (0xTxbi) [reply locksmith/package.json]: an oversight! fixed!

File: locksmith/src/config/config.ts [hunk 3]
```diff
@@ -50,6 +57,27 @@ Object.values(networks).forEach((network) => {
...
+// 5. Click "Create Service Account", enter a name and description, then click "Create" and "Continue".
+// 7. Go to the "Keys" tab, click "Add Key" > "Create New Key", choose "JSON", and click "Create" to download the key file.
+// 8. Set the environment variable to point to the key file:
+//    export GOOGLE_APPLICATION_CREDENTIALS="/path/to/your/service-account-file.json"
+// For more details, visit: https://cloud.google.com/docs/authentication/application-default-credentials
+*/
+
+const googleApplicationCredentials: Credentials = JSON.parse(
+  fs.readFileSync(
+    process.env.GOOGLE_APPLICATION_CREDENTIALS! as string,
```
[the remark this answers is not in the document]
Comment (0xTxbi) [reply locksmith/src/config/config.ts]: great! noted!

File: locksmith/src/operations/generate-pass/android/createPassObject.ts (hunk 2 above)
Comment (0xTxbi) [reply locksmith/src/operations/generate-pass/android/createPassObject.ts]: another oversight! all cleaned up!

**Comment by 0xTxbi** [comment]
> ok that looks good. I would love to see some unit tests here as well, especially to ensure that we have the right "format". I will set the env vars but please make sure you fix CI here

sure thing! necessary unit tests have been added for both pass generation operations and the controller.

CI has also been fixed!

**Pull request merged**